Add option to save request and response headers

malexmave opened the pull request:
Some use cases require getting access to request and response headers of page loads. This PR adds that feature. To enable it, set your `SCAN_MODULE_OPTIONS` in your config file to include `'RequestsExtractor.saveHeaders': True` for the `chromedevtools` scan module. Minimal config example:

```python
SCAN_MODULE_OPTIONS = {'chromedevtools': {'RequestsExtractor.saveHeaders': True}}
```

This PR also fixes a problem with loading a config file using the `--config file.py` option, where the scanner would crash due to a missing cast of the parameter string to a Path object.

malexmave committed:
Cast config_file to Path

The code later assumes that config_file is a path, however, it is often
passed as a string.  Fix this by always casting to a Path (if it is
already a path, the system will not care)

hprid commented:
Thank you for your pull request. Here are some remarks regarding the changes:

- Try to avoid cleanup commits like 5461bf7 in the same pull request. Instead rebase your code before and squash those commits into one if they are actually only one small change. `git rebase -i HEAD~3` or similar is your friend.
- When writing comments, focus on the _why_ instead of the _what_. Having a comment say that you cast something if the next code line is a cast does (5461bf7) not contribute to the understanding. I personally would leave out the comment, since it is absolutely fine to pass the config_file as an Path object; it is not required to be a string (although all calls are currently passing a string). Python will happily leave your Path object untouched if you “cast” it to Path again.
- Split logical changes into several pull requests, i.e., do not fix a minor bug together with a feature addition. I will not care for this pull request, but please keep it in mind for future ones.
- I would rename the option RequestsExtractor.saveHeaders to RequestsExtractor.save_headers to be consistent with Python's style guide, i.e., CamelCase for class names and snake_case for attributes.
- One remark regarding b411149, that is not relevant any more: Try to not include lines in a try block which do not raise a relevant exception that is meant to be caught.

After reworking your pull request should include two commits (config_file fix + RequestExtractors.save_headers option) that I will happily rebase into master.

malexmave committed:
Add option to capture HTTP headers

Sometimes, capturing the HTTP headers may be interesting (e.g. when the
referrer header for forwards should be investigated).  This commit
adds an option to save the headers for each request-response pair.

malexmave commented:
Thanks for the feedback. I rebased and squashed the relevant commits and changed the parameter name.
